src/engine/ruleset/decoders/redis/redis.yml

```diff
@@ -0,0 +1,110 @@
+
+# Example redis logs:
+
+# Darwin
+# 4961:M 30 May 12:50:13.457 * Increased maximum number of open files to 10032 (it was originally set to 4864).
+#                 _._
+#            _.-``__ ''-._
+#       _.-``    `.  `_.  ''-._           Redis 3.0.2 (00000000/0) 64 bit
+#   .-`` .-```.  ```\/    _.,_ ''-._
+#  (    '      ,       .-`  | `,    )     Running in standalone mode
+#  |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
+#  |    `-._   `._    /     _.-'    |     PID: 4961
+#   `-._    `-._  `-./  _.-'    _.-'
+#  |`-._`-._    `-.__.-'    _.-'_.-'|
+#  |    `-._`-._        _.-'_.-'    |           http://redis.io
+#   `-._    `-._`-.__.-'_.-'    _.-'
+#  |`-._`-._    `-.__.-'    _.-'_.-'|
+#  |    `-._`-._        _.-'_.-'    |
+#   `-._    `-._`-.__.-'_.-'    _.-'
+#       `-._    `-.__.-'    _.-'
+#           `-._        _.-'
+#               `-.__.-'
+# 4961:M 30 May 12:50:13.463 # Server started, Redis version 3.0.2
+
+# Debian
+# 30 May 10:04:45 . 0 clients connected (0 slaves), 618932 bytes in use, 0 shared objects
+
+# Windows
+# [2932] 31 May 04:32:08 - 0 clients connected (0 slaves), 1179968 bytes in use
+
+parse:
+  logpar:
+    # 26571:M 27 Dec 2018 11:19:18.874 * Synchronization with replica 10.114.208.18:6023 succeeded
+    - event.original: <process.pid>:<~log.role> <@timestamp/%d %b %Y %T> <log.level> <message>
+
+    # Darwin
+    - event.original: <process.pid>:<~log.role> <@timestamp/%d %b %T> <log.level> <message>
+
+    # Debian
+    - event.original: <@timestamp/%d %b %T> <log.level> <message>
+
+    # Windows
+    - event.original: <~/ignore/[><~log.pid>] <@timestamp/%d %b %T> <log.level> <message>
+
+# TODO: Could not map field 'log.offset' due to inability to persist values into yml variables.
+# Also, there is no helper that allows obtaining the size of an event.
+normalize:
+  - map:
+      - event.kind: event
+      - event.module: redis
+      - event.type: +a_append/info
+      - event.category: +a_append/database
+      - event.dataset: redis.log
+      - fileset.name: log
+      - input.type: log
+      - service.type: redis
+  - check:
+      - log.level: +ef_exists
+      - log.level: +s_eq/.
+    map:
+      - log.level: debug
+  - check:
+      - log.level: +ef_exists
+      - log.level: +s_eq/-
+    map:
+      - log.level: verbose
+  - check:
+      - log.level: +ef_exists
+      - log.level: +s_eq/*
+    map:
+      - log.level: notice
+  - check:
+      - log.level: +ef_exists
+      - log.level: +s_eq/#
+    map:
+      - log.level: warning
+  - check:
+      - ~log.role: +ef_exists
+      - ~log.role: +s_eq/M
+    map:
+      - redis.log.role: master
+  - check:
+      - ~log.role: +ef_exists
+      - ~log.role: +s_eq/S
+    map:
+      - redis.log.role: slave
+  - check:
+      - ~log.role: +ef_exists
+      - ~log.role: +s_eq/C
+    map:
+      - redis.log.role: child
+  - check:
+      - ~log.role: +ef_exists
+      - ~log.role: +s_eq/X
+    map:
+      - redis.log.role: sentinel
```

Comment:
"sentinel" literal is assigned to redis.log.role field
In other existing decoders the +s_append helper function is used. 
      - redis.log.role : +s_append/sentinel
Is a preferred way to do this?

Reply:
Missed this is only used for json objects. Disregard. Thank you.